src/Microsoft.AspNetCore.Mvc.Formatters.Json/Internal/MediaTypeHeaderValues.cs

```diff
@@ -15,5 +15,8 @@ internal class MediaTypeHeaderValues
 
         public static readonly MediaTypeHeaderValue ApplicationJsonPatch
             = MediaTypeHeaderValue.Parse("application/json-patch+json").CopyAsReadOnly();
```

Comment:
Currently JsonInputFormatter is registered before the JsonPatchInputFormatter. With these PR changes probably that order has to be reversed? 
https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Formatters.Json/Internal/MvcJsonMvcOptionsSetup.cs#L63-L74

(I am a bit surprised that we did not(based on GitHub search :-)) have E2E jsonpatch functional tests in this repo...I know Entroy repo has but we should have in this repo too...I will file a bug for this and work on it)